packages/cinterop/src/commonMain/kotlin/io/realm/internal/util/PlatformUtils.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.util
+
+/**
+ * Method to freeze state.
+ * Calls the platform implementation of 'freeze' on native, and is a noop on other platforms.
+ *
+ * Note, this method refers to Kotlin Natives notion of frozen objects, and not Realms variant
+ * of frozen objects.
+ */
+expect fun <T> T.freezeOnOldMM(): T
```

Comment:
We already have `io.realm.internal.platform.SystemUtils` which has these kind of methods. I would move it there instead of creating another file.

Reply:
The `SystemUtils` live in `library-base` but we need the `freezeOnOldMM` to be available in `cinterop`. I could move all the `SystemUtils` to `cinterop` if you'd like?